nm_vllm_utils/benchmarks/benchmark_serving.py

```diff
@@ -0,0 +1,772 @@
+                else:
+                    raise ValueError(
+                        "Invalid metadata format. Please use KEY=VALUE format."
+                    )
+
+        # Traffic
+        result_json["request_rate"] = (
+            args.request_rate if args.request_rate < float("inf") else "inf"
+        )
+
+        # Merge with benchmark result
+        result_json = {**result_json, **benchmark_result}
+
+        # Save to csv file
+        base_model_id = model_id.split("/")[-1]
+        file_name = f"{backend}-qps-{base_model_id}.csv"  # noqa
+        if args.result_dir:
+            file_name = os.path.join(args.result_dir, file_name)
+
+        if not os.path.exists(file_name):
+            with open(file_name, "w", newline="") as outfile:
+                writer = csv.DictWriter(outfile, fieldnames=result_json.keys())
+                writer.writeheader()
+
+        with open(file_name, "a", newline="") as outfile:
+            writer = csv.DictWriter(outfile, fieldnames=result_json.keys())
+            writer.writerow(result_json)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Benchmark the online serving throughput."
+    )
+    parser.add_argument(
+        "--backend",
+        type=str,
+        default="vllm",
+        choices=list(ASYNC_REQUEST_FUNCS.keys()),
+    )
+    parser.add_argument(
+        "--base-url",
+        type=str,
+        default=None,
+        help="Server or API base url if not using http host and port.",
+    )
+    parser.add_argument("--host", type=str, default="localhost")
+    parser.add_argument("--port", type=int, default=8000)
+    parser.add_argument(
+        "--endpoint",
+        type=str,
+        default="/v1/completions",
+        help="API endpoint.",
+    )
+    parser.add_argument(
+        "--dataset",
+        type=str,
+        default=None,
+        help="Path to the ShareGPT dataset, will be deprecated in the " "next release.",
+    )
+    parser.add_argument(
+        "--dataset-name",
+        type=str,
+        default="sharegpt",
+        choices=["sharegpt", "sonnet"],
+        help="Name of the dataset to benchmark on.",
+    )
+    parser.add_argument(
+        "--dataset-path", type=str, default=None, help="Path to the dataset."
+    )
+    parser.add_argument(
+        "--model",
+        type=str,
+        required=True,
+        help="Name of the model.",
+    )
+    parser.add_argument(
+        "--tokenizer",
+        type=str,
+        help="Name or path of the tokenizer, if not using the default tokenizer.",
+    )
+    parser.add_argument(
+        "--best-of",
+        type=int,
+        default=1,
+        help="Generates `best_of` sequences per prompt and " "returns the best one.",
+    )
+    parser.add_argument("--use-beam-search", action="store_true")
+    parser.add_argument(
+        "--num-prompts",
+        type=int,
+        default=1000,
+        help="Number of prompts to process.",
+    )
```

Comment:
I was comparing to the nm-vllm repo benchmark_serving file to try and figure out the main differences/additions, and noticed that there is a [shartgpt-output-len](https://github.com/neuralmagic/nm-vllm/blob/main/benchmarks/benchmark_serving.py#L529-L534) argument there as well. Is this file going to be the new source for the benchmark_serving file? Or is this something separate? Just want to make sure we don't lose anything that was added there recently